src/inventory.cpp

```diff
@@ -705,34 +705,32 @@ void InventoryList::moveItemSomewhere(u32 i, InventoryList *dest, u32 count)
 	}
 }
 
-u32 InventoryList::moveItem(u32 i, InventoryList *dest, u32 dest_i,
+ItemStack InventoryList::moveItem(u32 i, InventoryList *dest, u32 dest_i,
 		u32 count, bool swap_if_needed, bool *did_swap)
 {
+	ItemStack moved;
 	if (this == dest && i == dest_i)
-		return count;
+		return moved;
 
 	// Take item from source list
-	ItemStack item1;
-	if (count == 0)
-		item1 = changeItem(i, ItemStack());
-	else
-		item1 = takeItem(i, count);
```

Comment:
Occurrence 1: `IMoveAction::apply`. Value 0 is not possible due to previous guards.
Occurrence 2: ` IMoveAction::clientApply` Value 0 is not possible because empty itemstacks cannot be moved in the GUI.

Reply:
Could you update this comment (remove "0 for everything") and remove the default value of the `count` argument? https://github.com/minetest/minetest/blob/84e5866f03974360f9a73df1023c855eadefe936/src/inventory.h#L268-L272

Reply:
:+1: Done.